pages/components/ChatBox.jsx

```diff
@@ -0,0 +1,330 @@
+  useEffect(() => {
+    scrollToBottom()
+  }, [messageList])
+
+
+  const submitForm = event => {
+    event.preventDefault();
+    sendMessage(socketState, mObj);
+
+    if(mObj.message.length>=1){
+      setMessageList([...messageList,mObj])
+      setmObj({message:""})
+    }
+
+  }
+  const dummy = () => {};
+
+  return (
+    <>
+    <LogoWrapper >
+      <h3 style={{    paddingTop: '50px'}}>
+        Chat <span>Engo</span>
+      </h3>
+      <br />
+      <h3 style={{color: '#FA8072', fontSize:'15px'}}>
+      User: &nbsp; 
+        {context.user.name}
+      </h3>
+    </LogoWrapper>
+
+    <OuterDiv>
+
+      <ContactColumn style={{border:'1px solid black', 'border-radius':'10px', marginRight:'10px', marginLeft:'10px'}}> 
+        <div style={{display: 'block'}} >
+          <ContactListHeader>
+            <ContactListTitle>Active Users</ContactListTitle>
+          </ContactListHeader>
+          <ContactListColumn>
+            <ContactList>
+              { users?.map( user  => {
+              
+                if(user.name !== context.user.name){
+                return (
+                    <ContactListItem 
+                      key={dummy()}
+                      className={selectedContact === user.name? 'active': ''} 
+                      onClick={() => setSelectedContact(user.name)}>
+                        {user.name}
+                    </ContactListItem>
+                  )}   
+              })
+              }
+            </ContactList>
+          </ContactListColumn>
+        </div>      
+      </ContactColumn>
+      
+      <ChatColumn style={{border:'1px solid black', 'border-radius':'10px'}}>
+        <MessageListHeader>
+          <MessageListTitle>Your Messages </MessageListTitle>
+        </MessageListHeader>
+        <MessageListColumn>
+          <div>
+            {
+              messageList?.map( message => {
+                if(message.from===selectedContact || (message.from === context.user.name && message.to=== selectedContact))
+                return <div key={dummy()} style={{overflow: 'hidden'}}>
+                  { message.from=== context.user.name ?
+                  <Message style={{float:'right',backgroundColor:'#E1EBEE'}}>{ message.message }<span>{message.time}</span></Message>
+                  :
+                  <Message style={{float:'left', backgroundColor:'white'}}>{ message.message }<span>{message.time}</span></Message>
+                }
+                  </div>
+              })
+            }
+            <div ref={messagesEndRef} />
+          </div>
+        </MessageListColumn>
+        <div >
+          {
+            selectedContact?
+          <MessageForm onSubmit={submitForm}>
+            <Col9>
+              <MessageInput type="text" value={mObj.message} onChange={event => setmObj({
+                message:event.target.value,
+                to:selectedContact,
+                from:context.user.name,
+                time:new Date().toLocaleTimeString([], 
```

Comment:
why not set these params before sending messages instead of setting them again and again on each character change?